docs/docs/surface/algorithms/embedding.md

```diff
@@ -0,0 +1,164 @@
+
+    When a metric is read from UV coordinates (e.g., in an OBJ file), lengths should be equal on opposite sides of each seam edge.  These lengths will automatically be averaged if they are not exactly equal.  Also be careful that coordinates written to disk are not truncated to a small number of digits, which can make flat vertices appear to have slight negative curvature.
+
+The basic idea of Bobenko and Izmestiev's algorithm is to create an intrinsic tetrahedral mesh, where every triangle of the given surface mesh is connected to a central "apex" by radial edges that are initially very long, creating long skinny tetrahedra.  The radii of these edges are then optimized until the dihedral angle sum around all of them is equal to 2π, allowing the tetrahedra to be laid out in three-dimensional space without overlap.  During optimization, edge flips are applied to ensure that edges of the boundary (i.e., the surface mesh) remain convex rather than concave.
+
+Note that the Geometry Central implementation of this algorithm is based in part on the [Java reference implementation](https://gitlab.discretization.de/sechel/f1software/-/tree/master/src/alexandrov) by Stefan Sechelmann.  However, our implementation makes some significant modifications not advocated by the original authors.  Users interested in making comparisons (e.g., for academic research papers) are advised to consult the reference implementation.
+
+### Basic Usage
+
+The easiest way to compute an embedding is through "one shot" wrapper functions that take some description of the intrinsic metric as input, and produce an embedded mesh as output.
+
+Example
+```cpp
+#include "geometrycentral/surface/meshio.h"
+#include "geometrycentral/surface/embed_convex.h"
+
+// Load a mesh with UVs
+// Note: vertex positions will be ignored (e.g., can all be set to zero),
+// since our goal is to solve for vertex positions that exhibit the UV lengths.
+std::unique_ptr<SurfaceMesh> mesh;
+std::unique_ptr<VertexPositionGeometry> geometry;
+std::unique_ptr<CornerData<Vector2>> uvs;
+tie(mesh, geometry, uvs) = readParameterizedManifoldSurfaceMesh(filename);
+
+// Compute the embedding
+EmbedConvexResult result = embedConvex( *mesh, *uvs );
+
+if( result.success ) {
+    // do something with result.mesh and result.geometry
+}
+```
+
+Additional options can be passed to the embedder by passing an optional `EmbedConvexOptions` struct to the embedding function:
+
+```cpp
+EmbedConvexOptions options;
+options.tolerance = 1e-2; // use a looser tolerance than default
+EmbedConvexResult result = embedConvex( *mesh, *uvs, options );
+```
+
+??? func "`#!cpp EmbedConvexResult embedConvex(ManifoldSurfaceMesh& mesh, EdgeData<double>& edgeLengths, EmbedConvexOptions options = defaultEmbedConvexOptions);`"
+
+    Embed a metric described by a set of edge lengths.  Options are passed as a [RemeshOptions](#options) object.  If embedding fails, the flag `EmbedConvexResult::success` will be set to `false`.
```

Comment:
Might be useful to indicate in these docs whether the input mesh/edge lengths are changed, or if they are left unchanged. (Judging by the other docs I assume they are unchanged, and the `EmbedConvexResult` holds the solution.)

Another option could be to mark the inputs as `const`,  but it might not be that easy as some of the other geometry-central container methods are not marked const even though you'd think they would be, because data structures get automatically/lazily resized/initialized under the hood.

Reply:
Yeah, I had actually written the arguments as const at first but hit some compiler errors and didn't want to do meticulous constifying. Adding a note to the header/docs is a good idea.